examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py

```diff
@@ -52,8 +59,11 @@ def train(args, model, device, train_loader, optimizer, epoch_idx, core_context)
         loss = F.nll_loss(output, target)
         loss.backward()
         optimizer.step()
+        # Docs snippet end before: include core context
 
-        # NEW: Increment batch_idx by 1 in if statement to avoid reporting metrics at batch_idx 0.
+        # Docs snippet start after: increment batch idx
```

Comment:
This snippet is not used, and also the word "increment" means to "change a variable's value by adding 1 to it".  So this isn't incrementing `batch_idx` because it's not modifying `batch_idx`'s value, it's just using `batch_idx+1` instead of using `batch_idx` directly.

(I know this is not your fault).

Really what this should do is it should save a variable called `batches_completed` which is `batch_idx+1` and it should use that throughout the rest of this function:


```python
batches_completed = batch_idx + 1
if batches_completed % .args.log_interval = 0:
    ...
    core_context.train.report_training_metrics(
        steps_completed=batches_completed + epoch_idx * len(train_loader),
        ...
    )
```

You don't need to make the code change if you don't feel comfortable doing it, but please do replace the word "increment" in the comments.

Reply:
code edited; "increment" removed; replaced comment

# NEW: Increment batch_idx by 1 in if statement to avoid reporting metrics at batch_idx 0.

with:

        # NEW: Print training progress and loss at specified intervals
        # starting from the first batch.

Reply:
resolved